server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -354,6 +358,167 @@ public void testRegisterHandler() throws Exception {
 
     }
 
+    private static class Example implements NamedWriteable {
+        public static final String INVALID_NAME = "invalid_name";
+        public static final String NAME = "example";
+        private final String message;
+
+        Example(String message) {
+            this.message = message;
+        }
+
+        Example(StreamInput in) throws IOException {
+            this.message = in.readString();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(message);
+        }
+
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Example that = (Example) o;
+            return Objects.equals(message, that.message);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(message);
+        }
+    }
+
+    public void testGetNamedWriteables() throws Exception {
+        Path extensionDir = createTempDir();
+        List<String> extensionsYmlLines = Arrays.asList(
+            "extensions:",
+            "   - name: firstExtension",
+            "     uniqueId: uniqueid1",
+            "     hostName: 'myIndependentPluginHost1'",
+            "     hostAddress: '127.0.0.0'",
+            "     port: '9300'",
+            "     version: '0.0.7'",
+            "     description: Fake description 1",
+            "     opensearchVersion: '3.0.0'",
+            "     javaVersion: '14'",
+            "     className: fakeClass1",
+            "     customFolderName: fakeFolder1",
+            "     hasNativeController: false"
+        );
+        Files.write(extensionDir.resolve("extensions.yml"), extensionsYmlLines, StandardCharsets.UTF_8);
+        ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
+        transportService.start();
+        transportService.acceptIncomingRequests();
+        extensionsOrchestrator.setTransportService(transportService);
+
+        try (MockLogAppender mockLogAppender = MockLogAppender.createForLoggers(LogManager.getLogger(ExtensionsOrchestrator.class))) {
+
+            mockLogAppender.addExpectation(
```

Comment:
Let's test using assertStatements and actual method call instead mocking log messages here and for the rest